collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java

```diff
@@ -149,12 +165,67 @@ public void addTo(
     public void onAttach(@NonNull Context context) {
         super.onAttach(context);
         DaggerUtils.getComponent(context).inject(this);
+
+        mapFragmentDelegate = new MapFragmentDelegate(createConfigurator(), settingsProvider.getUnprotectedSettings(), this::onConfigChanged);
     }
 
     @Override
     public void onStart() {
         super.onStart();
-        mapProvider.onMapFragmentStart(this);
+        mapFragmentDelegate.onStart();
+    }
+
+    @NonNull
+    private MapConfigurator createConfigurator() {
+        org.odk.collect.shared.settings.Settings settings = settingsProvider.getUnprotectedSettings();
+        String basemap = settings.getString(KEY_BASEMAP_SOURCE);
+
+        MapConfigurator cftor = new OsmDroidMapConfigurator(
```

Comment:
The same here it could be placed in a separate class. Current map fragments are big enough.

Reply:
Agreed. This all lived in `MapProvider` before. I'm thinking we should switch Mapbox to this approach and then work out what the relationship between the `MapFragment` and `MapConfigurator` should look like. What do you think?